<html>

<head>
<meta charset="utf-8"/>
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Application to Biology</title>

<script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
</script>
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}});
</script>
<script type="text/javascript" src="/scripts/library.js"></script>
<script type="text/javascript" src="overrides.js"></script>
<link href="style.css" rel="stylesheet" type="text/css">
</head>

<body onresize="resizing()">


<div class="header">
<h1 style="color: rgb(165, 42, 42);">Understanding the World Through Code</h1>
<h2 style="color: 'black';">Funded through the NSF Expeditions in Computing Program</h2>
<div class="logos">
<img title='mit' src='images/mit.png' height='100%' />
<img title='ut' src='images/ut.png' height='100%'/>
<img title='caltech' src='images/caltech.png' height='100%'/>
<img title='rice' src='images/rice.png' height='100%'/>
<img title='penn' src='images/penn.png' height='100%'/>
<img title='stanford' src='images/stanford.png' height='100%'/>
</div>
<div id='navicon.div' class='navicon'>
<img id = 'navicon' src="images/burger.rest.png" 
onmouseenter="iconEnter(this, 'burger')"
onmouseleave="iconLeave(this, 'burger')"
onclick = 'toggleNav()'
title="Navigation" width="34pt"></img>
</div>
</div>


<div id="navigator" class="sidenav">
<a href="#about">Lecture 1</a> 
<a href="#services">Lecture 2</a> 
<a href="#clients">Lecture 3</a> 
<a href="#contact">Lecture 4</a>
</div>
<script type="text/javascript">
loadNavBar();
</script>

<div class="content">

<h1>RNA Splicing</h1>

<iframe class="slideview" style="padding:0; border:10px solid #ffffff;" 
        name="version3"
src="https://code.playskript.com/view.html?id=1ad7ec05f5304c49951b5c88aa5762f9d6863da25d2b46db69"
sandbox="allow-forms
allow-pointer-lock
allow-same-origin
allow-scripts
allow-top-navigation"
height="300"
width="500">
  <p>Your browser does not support iframes.</p>
</iframe>

<p>
In most bacteria, the process of protein synthesis involves a transcription step, 
where a strand of messenger RNA is assembled as a copy of a gene with 
the help of RNA polymerase, followed by 
a translation step, where Rhybosomes decode the gene into a sequence of aminoacids 
that will fold into a protein. 

Back in the 1970s, however, co-PI Phillip Sharp and his team discovered that in 
eukaryotes, transcription also involves <i>splicing</i>, where a complex of 
molecules 
called the <i>spliceosome</i> would bind to the RNA to remove segments of 
non-coding RNA known as <i>introns</i>, leaving behind the expressed portions 
of the RNA strand known as <i>exons</i>.    
</p>

<p>
In the years since that discovery, biology has learned a great amount about the 
mechanisms involved in RNA splicing and the myriad of RNA-binding proteins 
that regulate the action of the splyceosome. However, we are still far from a
comprehensive model that would help us predict with certainty the effect that
different intervations---whether mutations or the addition of small molecules---  
will have on how a given RNA sequence will be spliced.   
</p>

<i><a href="splicingPrimer.html">[Check out our quick primer to learn more about splicing.]</a></i>

<p>
Developing better models of splicing is critical to understanding 
a myriad of diseases. For example, many genetic diseases including 
early onset Parkinson disease, spinal muscular atrophy and 
amyotrophic lateral sclerosis can all be traced back to mutations 
that alter splicing. Splicing errors are also behind many forms of 
cancer.  
</p>

<p>
Deep learning has already demonstrated some promise in turning the large amounts
of data available about how particular genes are spliced into predictions
about the effect of certain mutations on RNA splicing. However, the neurosymbolic 
learning techniques that we plan to develop as part of this proposal could allow us 
to develop richer models that capture more of what is known about the splicing process 
and the molecules involved. This could help us move beyond simply making predictions,
to a real understanding of <i>why</i> particular mutations have the effects that they 
have. 
</p>

<p>
We have made significant progress on this project, completing our first major objective:
decoupling the identification of RBP binding sites from the prediction of splicing.

Specifically, we have done this via the "Adjusted Motifs" approach, as depicted in the presentation below.
</p>

<iframe class="alwaysbig" style="padding:0; border:10px solid #ffffff;"
  src="https://code.playSkript.com/view.html?id=c073a4f056a0b8d9563e58d05fa8c67a1c4428e3158fb4e6cc" sandbox="allow-forms
allow-pointer-lock
allow-same-origin
allow-scripts
allow-top-navigation" height="500" width="500">
  <p>Your browser does not support iframes.</p>
</iframe>

<p>New research from our group <cite>gupta2023improved-ours</cite> aims to better understand how cells process genetic information to make proteins.
  Specifically, the model models splicing as a two-stage process, first modelling the attachment of RNA-binding proteins, and then modeling
  splicing based on the locations of these attachments. This improves interpretability by allowing researchers to see which RNA-binding proteins
  are predicted to attach where, and to delete specific ones to see how splicing predictions change.

<p> Additionally, the project yielded improved models of RBP attachment, as validated using several experiments.

<p>The key findings are:


<ul>
<li>The model greatly improved its ability to predict where RNA-binding proteins will attach compared to just using the test tube measurements. This suggests it learned features relevant to how binding happens inside cells.</li>
<li>The model improved substantially over prior RBP-based interpretable models of splicing, gaining 80% accuracy</li>
<li>Unlike previous "black box" models, this model allows interpretability. The researchers can see which RNA-binding proteins it predicts will attach where, and delete specific ones to see how splicing predictions change.</li>
<li>The model largely learned an appropriate pattern of activity for different classes of RNA-binding proteins based on their known biology.</li>
</ul>

<p>Overall, this interpretable model integrates both lab measurements and genomic data to better reveal the "splicing code" - the specific rules cells use to process RNA messages into proteins.
In the future, this can be used for research and clinical applications.



<script>
processDocument();
</script>

<div class="footnotes">

<footnotes>

</footnotes>

</div>
</div>

</body>



</html>
